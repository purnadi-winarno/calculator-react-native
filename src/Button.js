import React from "react";
import { View, TouchableOpacity, Text } from "react-native";
import styles from "./styles";

const Button = ({ text, square, onPressHandler }) => {
  //console.log("button rendered"); //ensure it's not re-rendered when parent's props changes
  const buttonStyle = square ? styles.squareButton : styles.rectangleButton;
  return (
    <TouchableOpacity
      style={[styles.button, buttonStyle]}
      onPress={() => onPressHandler(text)}
    >
      <Text style={styles.buttonText}>{text}</Text>
    </TouchableOpacity>
  );
};

export default React.memo(Button);
