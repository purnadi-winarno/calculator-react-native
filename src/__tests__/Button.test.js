import { shallow } from "enzyme";
import * as React from "react";
import Button from "../Button";

const setUpComponent = (props = {}) => shallow(<Button {...props} />);

describe("Testing button", () => {
  let wrapper;
  let spy;

  beforeEach(() => {
    spy = jest.fn();
    const props = {
      text: "7",
      square: true,
      onPressHandler: spy
    };
    wrapper = setUpComponent(props);
  });

  it("renders square button correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders non square button correctly", () => {
    wrapper.setProps({ square: false });
    expect(wrapper).toMatchSnapshot();
  });

  it("check button action press handler", () => {
    wrapper
      .first()
      .props()
      .onPress();

    expect(spy).toBeCalled();
  });
});
