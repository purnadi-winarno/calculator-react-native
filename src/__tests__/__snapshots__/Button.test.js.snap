// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing button renders non square button correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#eee",
        "borderRadius": 5,
        "margin": 5,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
      }
    }
  >
    7
  </Text>
</TouchableOpacity>
`;

exports[`Testing button renders square button correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#eee",
        "borderRadius": 5,
        "margin": 5,
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
      }
    }
  >
    7
  </Text>
</TouchableOpacity>
`;
